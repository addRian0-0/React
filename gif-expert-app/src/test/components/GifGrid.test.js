import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('probando gifGrid', () => {

    const category = "Arcane";

    test('sacar snapshot jajaj q loco', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot();

    })

    test('debe mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const imgs = [{
            id: 'ABC',
            url: 'https://localhost/oli.png',
            title: 'Cualquier cosa'
        }, {
            id: 'ABC1',
            url: 'https://localhost/oli.png',
            title: 'Cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: imgs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe( imgs.length );

    })


})