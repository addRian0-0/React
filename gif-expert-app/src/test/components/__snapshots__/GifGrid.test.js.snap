// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando gifGrid debe mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Arcane
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/oli.png"
    />
    <GifGridItem
      id="ABC1"
      key="ABC1"
      title="Cualquier cosa"
      url="https://localhost/oli.png"
    />
  </div>
</Fragment>
`;

exports[`probando gifGrid sacar snapshot jajaj q loco 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Arcane
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
